File: src/main/java/dev/iancmd/operationrelife/init/ModDimensions.java
package dev.iancmd.operationrelife.init;

import net.minecraft.resources.ResourceKey;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.level.Level;
import net.minecraft.world.level.dimension.DimensionType;

import static dev.iancmd.operationrelife.core.OperationRelife.MODID;
import static dev.iancmd.operationrelife.core.OperationRelife.getLogger;
import static net.minecraft.core.registries.Registries.DIMENSION;
import static net.minecraft.core.registries.Registries.DIMENSION_TYPE;
import static org.apache.logging.log4j.Level.DEBUG;

public class ModDimensions {
  public static final ResourceKey<Level> TERMINAL_ISLE_KEY = ResourceKey.create(DIMENSION, new ResourceLocation(MODID, "terminal_isle"));
  public static final ResourceKey<DimensionType> TERMINAL_ISLE_TYPE = ResourceKey.create(DIMENSION_TYPE, new ResourceLocation(MODID, "terminal_isle"));
  public static void DimensionRegistry() {
    getLogger().log(DEBUG, "Registering dimensions...");
  }
}

File: src/main/kotlin/dev/iancmd/operationrelife/init/ModDimensions.kt
package dev.iancmd.operationrelife.init

import dev.iancmd.operationrelife.core.OperationRelife
import net.minecraft.core.registries.Registries
import net.minecraft.resources.ResourceKey
import net.minecraft.resources.ResourceLocation
import net.minecraft.world.level.dimension.DimensionType
import org.apache.logging.log4j.Level

object ModDimensions {
  val TERMINAL_ISLE_KEY = ResourceKey.create<net.minecraft.world.level.Level>(
    Registries.DIMENSION,
    ResourceLocation(OperationRelife.Companion.MODID, "terminal_isle")
  )
  val TERMINAL_ISLE_TYPE = ResourceKey.create<DimensionType>(
    Registries.DIMENSION_TYPE,
    ResourceLocation(OperationRelife.Companion.MODID, "terminal_isle")
  )

  fun dimensionRegistry() {
    OperationRelife.Companion.getLogger().log(Level.DEBUG, "Registering dimensions...")
  }
}
